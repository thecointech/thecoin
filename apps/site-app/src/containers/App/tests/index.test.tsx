import React from 'react';
import { createRenderer } from 'react-test-renderer/shallow';

const renderer = createRenderer();

// Mock the various API's we hook here
jest.mock('react-router', () => ({
  useLocation: () => { key: "TESTING"}
}));
jest.mock('@thecointech/shared/containers/AccountMap', () => ({
  useAccountMapStore: jest.fn(),
}));
jest.mock('@thecointech/shared/containers/FxRate/reducer', () => ({
  useFxRatesStore: jest.fn(),
}));
jest.mock('@thecointech/shared/containers/PageSidebar/reducer', () => ({
  useSidebar: jest.fn(),
}));

// This import can trigger a compilation of the entire app
// (with numerous side-effects) so we skip it entirely
jest.mock('../../MainRouter', () => ({
  MainRouter: () => <div>Router</div>
}));

import { App } from '../index';

describe('<App />', () => {
  it('should render and match the snapshot', () => {
    renderer.render(<App />);
    const renderedOutput = renderer.getRenderOutput();
    expect(renderedOutput).toMatchSnapshot();
  });
});
