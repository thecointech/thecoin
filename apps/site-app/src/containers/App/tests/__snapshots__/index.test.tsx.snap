// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<App /> should render and match the snapshot 1`] = `
<React.Fragment>
  <injectIntl(Component) />
  <Navigation />
  <Container
    style={
      Object {
        "backgroundColor": "#fffff f",
        "width": "100%",
      }
    }
  >
    <Unknown>
      <MainPageTransition>
        <section>
          <MainRouter />
        </section>
      </MainPageTransition>
    </Unknown>
  </Container>
  <Unknown />
</React.Fragment>
`;
